tools/windows/install-help/cal/FinalizeInstall.cpp

```diff
@@ -333,6 +340,40 @@ UINT doFinalizeInstall(CustomActionData &data)
         goto LExit;
     }
 
+#ifndef _CONSOLE
+    {
+        std::array<std::filesystem::path, 2> embeddedArchiveLocations =
+        {
+            installdir + L"\\embedded2.7z",
+            installdir + L"\\embedded3.7z",
+        };
+
+        for (const auto path : embeddedArchiveLocations)
+        {
+            if (std::filesystem::exists(path))
```

Comment:
Does `decompress_archive` erase the target directory if it already exists, or does it mix both the existing and new contents? If it's the second case, I think we need to delete the existing path before extracting.

Eg: Imagine the scenario where a file from an integration disappears in a new version of the Agent, eg: Agent 7.33 contains `embedded2/my_integration/file.py` but 7.34 doesn't contain that file. When we upgrade to 7.34 and decompress `embedded2.7z`, the `file.py` from 7.33 will remain in the filesystem?

Also, is this something an automated test could validate?

Reply:
Makes sense. If the upgrade first deletes everything this shouldn't be a problem. It's something that would be nice to have an automated test for, if we don't have one already, though.